components/Playlist/PlaylistButton.tsx

```diff
@@ -0,0 +1,73 @@
+import { Dimensions, Pressable, View } from "react-native";
+import { Dialog } from "@rneui/themed";
+import { MaterialCommunityIcons } from "@expo/vector-icons";
+import { useTheme } from "@react-navigation/native";
+import { useAuth } from "@/hooks";
+import { useState } from "react";
+import { brandColors } from "@/constants";
+import { Button } from "@/components/Button";
+import { Text } from "@/components/Text";
+import { BadgeIcon } from "@/components/BadgeIcon";
+import { ChoosePlaylistButton } from "./ChoosePlaylistButton";
+import { CreatePlaylistButton } from "./CreatePlaylistButton";
+
+interface PlaylistButtonProps {
+  size: number;
+  contentId: string;
+  isMusic: boolean;
+}
+
+export const PlaylistButton = ({
+  size,
+  contentId,
+  isMusic,
+}: PlaylistButtonProps) => {
+  const { pubkey } = useAuth();
+  const { colors } = useTheme();
+  const screenWidth = Dimensions.get("window").width;
+  const [isDialogOpen, setIsDialogOpen] = useState(false);
+
+  return pubkey && isMusic ? (
+    <View style={{ backgroundColor: colors.background }}>
```

Comment:
nit - can use early return here too
```suggestion

  if (!pubkey || !isMusic) return;
  
  return (
    <View style={{ backgroundColor: colors.background }}>
```